repl/src/main/scala/ammonite/repl/interp/ClassLoaders.scala

```diff
@@ -121,11 +123,42 @@ class SpecialClassLoader(parent: ClassLoader, parentHash: Array[Byte])
     * classloader can get at them.
     */
   val newFileDict = mutable.Map.empty[String, Array[Byte]]
+  val classOutputDir = System.getProperty("ammonite.repl.class.outputdir", null)
+
   def findClassPublic(name: String) = findClass(name)
   val specialLocalClasses = Set(
     "ammonite.repl.frontend.ReplBridge",
     "ammonite.repl.frontend.ReplBridge$"
   )
+
+  def putClassBytes(name: String, bytes: Array[Byte]) = {
+    newFileDict(name) = bytes
+    if (classOutputDir != null) {
+      // Create the tmp directory and save class file there.
+      val f = new File(classOutputDir)
```

Comment:
Since we're inside Ammonite, let's using Ammonite-Ops to do this:

```scala
import ammonite.ops._
mkdir(classOutputDir)
write(classOutputDir/s"$name.class", bytes)
```

That handles all the folder-already-exists cases etc. automatically